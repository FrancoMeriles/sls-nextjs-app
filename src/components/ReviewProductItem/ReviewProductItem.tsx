import { FC } from 'react'
import { Box, Text, Flex } from '@chakra-ui/react'

// Icons
import { Thumbs } from '@base/icons'

// Components
import RatedStar from '@base/components/RatedStar'

interface ReviewProductItemProps {
  review: ReviewInterface
}

interface ReviewInterface {
  id: number
  rate: number
  title: string
  content: string
  likes: number
  dislikes: number
}

const ReviewProductItem: FC<ReviewProductItemProps> = ({ review }) => {
  return (
    <Box mt={4} mb={6}>
      <RatedStar rateAverage={review.rate} boxSize={4} />
      <Text fontSize="sm" fontWeight="semibold" mt={3} mb={3}>
        {review.title}
      </Text>
      <Text fontSize="sm">{review.content}</Text>
      <Flex alignItems="center" mt={4}>
        <Thumbs mr={2} mt={1} /> <Text fontSize="sm">{review.likes}</Text>
        <Thumbs transform="rotate(180deg)" mr={2} mt={1} ml={5} />
        <Text fontSize="sm">{review.dislikes}</Text>
      </Flex>
    </Box>
  )
}

export default ReviewProductItem
