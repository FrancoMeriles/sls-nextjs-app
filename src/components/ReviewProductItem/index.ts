import ReviewProductItem from './ReviewProductItem'

export default ReviewProductItem
