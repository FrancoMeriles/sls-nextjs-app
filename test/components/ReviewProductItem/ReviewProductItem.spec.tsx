import { render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'

import ReviewProductItem from '@base/components/ReviewProductItem'

const review = {
  id: 123456,
  rate: 4,
  title: '__TITLE_ONE__',
  content: '__CONTENT_ONE__',
  likes: 3,
  dislikes: 1,
}

describe('ReviewProductItem', () => {
  it('should render and match snapshot with required props', () => {
    const tree = renderer.create(<ReviewProductItem review={review} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should renders and display the correct data', async () => {
    render(<ReviewProductItem review={review} />)
    expect(screen.getByText(review.title)).toBeInTheDocument()
    expect(screen.getByText(review.content)).toBeInTheDocument()
  })
})
