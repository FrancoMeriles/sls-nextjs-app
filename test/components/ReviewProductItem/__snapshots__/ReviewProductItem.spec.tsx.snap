// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewProductItem should render and match snapshot with required props 1`] = `
<div
  className="css-1puzibz"
>
  <div
    className="css-0"
  >
    <svg
      className="chakra-icon css-1phmqac"
      focusable={false}
      paint="full"
      viewBox="0 0 10 10"
    >
      <path
        d="M5.056 8L1.931 9.648l.597-3.49L0 3.684l3.494-.509L5.056 0l1.562 3.176 3.494.51-2.528 2.471.597 3.491z"
        fill="currentColor"
      />
    </svg>
    <svg
      className="chakra-icon css-1phmqac"
      focusable={false}
      paint="full"
      viewBox="0 0 10 10"
    >
      <path
        d="M5.056 8L1.931 9.648l.597-3.49L0 3.684l3.494-.509L5.056 0l1.562 3.176 3.494.51-2.528 2.471.597 3.491z"
        fill="currentColor"
      />
    </svg>
    <svg
      className="chakra-icon css-1phmqac"
      focusable={false}
      paint="full"
      viewBox="0 0 10 10"
    >
      <path
        d="M5.056 8L1.931 9.648l.597-3.49L0 3.684l3.494-.509L5.056 0l1.562 3.176 3.494.51-2.528 2.471.597 3.491z"
        fill="currentColor"
      />
    </svg>
    <svg
      className="chakra-icon css-1phmqac"
      focusable={false}
      paint="full"
      viewBox="0 0 10 10"
    >
      <path
        d="M5.056 8L1.931 9.648l.597-3.49L0 3.684l3.494-.509L5.056 0l1.562 3.176 3.494.51-2.528 2.471.597 3.491z"
        fill="currentColor"
      />
    </svg>
    <svg
      className="chakra-icon css-1phmqac"
      focusable={false}
      paint="empty"
      viewBox="0 0 10 10"
    >
      <path
        d="M5.056 8L1.931 9.648l.597-3.49L0 3.684l3.494-.509L5.056 0l1.562 3.176 3.494.51-2.528 2.471.597 3.491z"
        fill="#e0e0e0"
      />
    </svg>
  </div>
  <p
    className="chakra-text css-160q44m"
  >
    __TITLE_ONE__
  </p>
  <p
    className="chakra-text css-g5hezf"
  >
    __CONTENT_ONE__
  </p>
  <div
    className="css-1ivb442"
  >
    <svg
      className="chakra-icon css-r96wal"
      focusable={false}
      viewBox="0 0 17 17"
    >
      <path
        d="M6.15 5.464l.86-.334L9.576 0l1.017.509a3 3 0 0 1 1.473 3.719l-.286.777h2.8a2.4 2.4 0 0 1 2.243 3.256l-1.205 3.159a4.2 4.2 0 0 1-4.682 2.635l-3.546-.65H6.15v1.2H2v-10.2h4.15v1.06zm0 1.288v5.453h1.294l.108.01 3.6.66a3 3 0 0 0 3.343-1.883L15.7 7.833a1.2 1.2 0 0 0-1.12-1.628h-4.52l.88-2.392a1.8 1.8 0 0 0-.829-2.202l-2.234 4.47-1.728.67zM3.2 13.404h1.75v-7.8H3.2v7.8z"
        fill="currentColor"
        fillRule="nonzero"
      />
    </svg>
     
    <p
      className="chakra-text css-g5hezf"
    >
      3
    </p>
    <svg
      className="chakra-icon css-10m0rwy"
      focusable={false}
      viewBox="0 0 17 17"
    >
      <path
        d="M6.15 5.464l.86-.334L9.576 0l1.017.509a3 3 0 0 1 1.473 3.719l-.286.777h2.8a2.4 2.4 0 0 1 2.243 3.256l-1.205 3.159a4.2 4.2 0 0 1-4.682 2.635l-3.546-.65H6.15v1.2H2v-10.2h4.15v1.06zm0 1.288v5.453h1.294l.108.01 3.6.66a3 3 0 0 0 3.343-1.883L15.7 7.833a1.2 1.2 0 0 0-1.12-1.628h-4.52l.88-2.392a1.8 1.8 0 0 0-.829-2.202l-2.234 4.47-1.728.67zM3.2 13.404h1.75v-7.8H3.2v7.8z"
        fill="currentColor"
        fillRule="nonzero"
      />
    </svg>
    <p
      className="chakra-text css-g5hezf"
    >
      1
    </p>
  </div>
</div>
`;
